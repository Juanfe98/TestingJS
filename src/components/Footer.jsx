import React from 'react';
import '../styles/components/Footer.styl';

const Footer = () => (
  <div className="Footer">
    <p className="Footer-title">Platzi Store</p>
    <p className="Footer-copy">Testing snapshot on Footer</p>
  </div>
);

export default Footer;
