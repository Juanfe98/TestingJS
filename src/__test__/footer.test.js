import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Footer from '../components/Footer';

it('Test', () => {
  expect('hola').toBe('hola');
});

describe('Footer Component tests', () => {
  const footer = mount(<Footer />);

  it('Render del componente Footer', () => {
    expect(footer.length).toEqual(1);
  });

  it('Should render the footer title', () => {
    expect(footer.find('.Footer-title').text()).toEqual('Platzi Store');
  });
});

describe('Footer Snapshot', () => {
  it('Component should match the UI snapshot version', () => {
    const footer = create(<Footer />);
    expect(footer.toJSON()).toMatchSnapshot();
  });
});
