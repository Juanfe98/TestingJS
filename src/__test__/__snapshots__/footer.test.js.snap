// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot Component should match the UI snapshot version 1`] = `
<div
  className="Footer"
>
  <p
    className="Footer-title"
  >
    Platzi Store
  </p>
  <p
    className="Footer-copy"
  >
    Testing snapshot on Footer
  </p>
</div>
`;
